File: library/src/main/java/com/getstream/sdk/chat/rest/EventHandler.java
package com.getstream.sdk.chat.rest;

import android.os.Handler;
import android.os.Message;

import com.getstream.sdk.chat.model.Event;


public class EventHandler extends Handler {
    private WebSocketService webSocketService;

    public EventHandler(WebSocketService webSocketService) {
        this.webSocketService = webSocketService;
    }

    @Override
    public void handleMessage(Message msg) {
        super.handleMessage(msg);
        webSocketService.getWebSocketListener().onWSEvent((Event) msg.obj);
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/socket/EventHandler.kt
package io.getstream.chat.android.core.poc.library.socket

import android.os.Handler
import android.os.Message
import io.getstream.chat.android.core.poc.library.events.ChatEvent

class EventHandler(private val webSocketService: StreamWebSocketService) : Handler() {
    override fun handleMessage(msg: Message) {
        super.handleMessage(msg)
        webSocketService.onWsEvent(msg.obj as ChatEvent)
    }
}